vql/tools/s3_upload.go

```diff
@@ -0,0 +1,127 @@
+//+build extras
+
+package tools
+
+import (
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"golang.org/x/net/context"
+	"www.velocidex.com/golang/velociraptor/glob"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+	"www.velocidex.com/golang/velociraptor/vql/networking"
+	"www.velocidex.com/golang/vfilter"
+)
+
+type S3UploadArgs struct {
+	File              string `vfilter:"required,field=file,doc=The file to upload"`
+	Name              string `vfilter:"optional,field=name,doc=The name of the file that should be stored on the server"`
+	Accessor          string `vfilter:"optional,field=accessor,doc=The accessor to use"`
+	Bucket            string `vfilter:"required,field=bucket,doc=The bucket to upload to"`
+	Region            string `vfilter:"required,field=region,doc=The region the bucket is in"`
+	CredentialsKey    string `vfilter:"required,field=credentialskey,doc=The AWS key credentials to use"`
+	CredentialsSecret string `vfilter:"required,field=credentialssecret,doc=The AWS secret credentials to use"`
+}
+
+type S3UploadFunction struct{}
+
+func (self *S3UploadFunction) Call(ctx context.Context,
+	scope *vfilter.Scope,
+	args *vfilter.Dict) vfilter.Any {
```

Comment:
This was refactored recently to ordereddict.Dict - it fails to build:

```
vql\tools\s3_upload.go:33:8: undefined: vfilter.Dict
```

Reply:
Sorry, forgot to sync against upstream. Will rebase and fix.